integration-snippets/README.md

```diff
@@ -0,0 +1,81 @@
+# Purpose
+
+The scripts and snippets here are intended to support automating the upgrade
+of a cluster through the addition of two initContainers and a Job. The scripts
+are added to the calico/upgrade container image so they can be ran as
+containers.
+
+# Configuration
+
+Each of the scripts need some configuration, here are the common options
+between the scripts.
+
+## V3 datastore configuration
+The following can be used to configure access to the V3 datastore.
+
+  CALICO_ETCD_ENDPOINTS
+  CALICO_ETCD_USERNAME
+  CALCIO_ETCD_PASSWORD
+  CALICO_ETCD_KEY_FILE
+  CALICO_ETCD_CERT_FILE
+  CALICO_ETCD_CA_CERT_FILE
+
+## V1 datastore configuration
+The following can be used to configure access to the V1 datastore.
+
+  CALICO_APIV1_ETCD_SCHEME
+  CALICO_APIV1_ETCD_AUTHORITY
+  CALICO_APIV1_ETCD_ENDPOINTS
+  CALICO_APIV1_ETCD_USERNAME
+  CALICO_APIV1_ETCD_PASSWORD
+  CALICO_APIV1_ETCD_KEY_FILE
+  CALICO_APIV1_ETCD_CERT_FILE
+  CALICO_APIV1_ETCD_CA_CERT_FILE
+
+## calico-upgrade configuration
+
+It may be useful to provide some additional options or args to calico-upgrade.
+The following are provided for that purpose:
+* `UPGRADE_OPTIONS` can be used to to pass log level to the commands.
+* `UPGRADE_ARGS` could be used to to pass apiconfig if config files are
+  available instead of enviornment variables.
+
+# Node Daemonset initContainer
+
+This container will be responsible for checking if the datastore needs to be
+migrated and if needed will start the migration. If the datastore migration is
+in progress this container will block startup until it is complete.
+The [script](node-init-container.sh) which does the above can be added to the
+calico-node Daemonset with something like
+[this snippet](node-init-container.yaml).
+
+# calico/kube-controller initContainer
+
+This container will be responsible for checking if the datastore needs to be
+migrated and block startup until the migration is finished.
```

Comment:
It seems just from reading the description that this container does everything the `Node Daemonset initContainer` does except for starting the migration itself. Is there a reason we need this?

Reply:
nvm, reading the rest of this, I see it is for the different components we have.

Reply:
It was a good thought. I've changed it a bit though and should have changes pushed soon, and it diverges a bit more.  What was there before worked but I think what I have now better matches the intended behavior.